sensors/generic-sensor-tests.js

```diff
@@ -259,6 +259,22 @@ function runGenericSensorTests(sensorType) {
      }
   }, `${sensorType.name}: throw 'NotSupportedError' for an unsupported sensor option`);
 
+  test(() => {
+    const invalidFreqs = [
+      "invalid",
+      NaN,
+      Infinity,
+      -Infinity,
+      {},
+      undefined
+    ];
+    invalidFreqs.map(freq => {
+      assert_throws(new TypeError(),
+                    () => { new sensorType({frequency: freq}) },
+                    `when freq is ${freq}`);
+    });
+  }, `${sensorType.name}: throw 'TypeError' if frequency is an incorrect value type`);
```

Comment:
The test name is not quite appropriate, as `Infinity` is also typeof `number`.  So I think we can just describe it as "throw 'TypeError' if frequency is invalid"

Reply:
Thx @Honry, updated.